### ISSUE TITLE
Add class imbalance weighting to loss functions

### ISSUE THREAD

**Issue by jeshraghian** [jeshraghian/snntorch#88]
Apply on/off target weighting to `snntorch.functional` losses in the same way the PyTorch enables weighting.

Cross Entropy-based losses should be straightforward; Mean Square Error Losses are less trivial.
